addons/event_crm_sale/models/sale_order.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    def _check_new_registration(self, registration):
+        """
+        This method is rewrite in website_event_crm_sale to avoid to create rules one by one when
+        an order is make on the website.
+        """
+        return True
+
+    def _update_registrations(self, confirm=True, cancel_to_draft=False, registration_data=None):
```

Comment:
I think this approach is quite expensive in term of search and code. Indeed here you perform a global search before, after, and try to make a multi-registrations record set based on difference. On a real production db this is very costly.

I would suggest another approach, what do you think of
```
  * define a main_registration_id field on registration model (directly in event);
  * this field will be set only on "childs" registrations, aka registrations after the first one (a mapping to their "parent");
  * set it manually in website_event and website_event_sale. As registrations are created in loops, it should be quite easy to get the ID of the first created one and update values given to other create calls;
 ```

You will have to update your main rule execution method to use this main_registration_id field to know if you are working on a batch or on a single record.

Normally with that you will not have to create bridge module, as website_event and website_event_sale will already set this field and base code will handle it.

I know there are some tricky entry points in event_sale with SO and the "update registration" wizard, to check a bit.